apps/web/app/api/v1/management/storage/local/route.ts

```diff
@@ -53,14 +54,23 @@ export const POST = async (req: NextRequest): Promise<Response> => {
 
   const session = await getServerSession(authOptions);
 
-  if (!session || !session.user) {
-    return responses.notAuthenticatedResponse();
-  }
-
-  const isUserAuthorized = await hasUserEnvironmentAccess(session.user.id, environmentId);
-
-  if (!isUserAuthorized) {
-    return responses.unauthorizedResponse();
+  if (!session) {
+    //check whether its using API key
+    const authentication = await authenticateRequest(req);
+    if (!authentication) return responses.notAuthenticatedResponse();
+    const isAuthorized = authentication.environmentPermissions.some(
```

Comment:
can we replace this check with the helper `hasPermission` like we do at other places? I think the helper function `hasPermission` is defined under the api-key directory but we can make this reusable and directly use that in both the api routes 🙏